File: src/main/kotlin/com/tezal/hadith/security/MyUserDetails.java
package com.tezal.hadith.security;

import com.tezal.hadith.entity.UserEntity;
import com.tezal.hadith.repo.UserRepo;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;


@Service
public class MyUserDetails implements UserDetailsService {

  private final UserRepo userRepository;

  public MyUserDetails(UserRepo userRepository) {
    this.userRepository = userRepository;
  }

  @Override
  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
    final UserEntity user = userRepository.findByUserName(username);

    if (user == null) {
      throw new UsernameNotFoundException("User '" + username + "' not found");
    }

    return org.springframework.security.core.userdetails.User//
        .withUsername(username)//
        .password(user.getPassword())//
        .authorities(user.getRoles())//
        .accountExpired(false)//
        .accountLocked(false)//
        .credentialsExpired(false)//
        .disabled(false)//
        .build();
  }

}

File: src/main/kotlin/com/tezal/hadith/security/MyUserDetails.kt
package com.tezal.hadith.security

import com.tezal.hadith.repo.UserRepo
import org.springframework.security.core.userdetails.UserDetailsService
import kotlin.Throws
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.User
import org.springframework.stereotype.Service

@Service
class MyUserDetails(private val userRepository: UserRepo) : UserDetailsService {
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        val (_, password, _, _, roles) = userRepository.findByUserName(username)
        return User //
                .withUsername(username) //
                .password(password) //
                .authorities(roles) //
                .accountExpired(false) //
                .accountLocked(false) //
                .credentialsExpired(false) //
                .disabled(false) //
                .build()
    }
}